### PR TITLE
Share: Send data to other apps

### PR DESCRIPTION
**Pull request by yaraki**
Change-Id: Ie68d0e8431102eda3131739e9482d8ba1988a44e

### CODE REVIEW

**Review by JolandaVerhoef — APPROVED** [review]

File: samples/README.md [hunk 1]
```diff
@@ -88,6 +88,8 @@ This samples shows how to use the BLE intent to scan for devices
 This sample shows how to detect that the user capture the screen in Android 14 onwards
 - [Selected Photos Access](storage/src/main/java/com/example/platform/storage/mediastore/SelectedPhotosAccess.kt):
 Check and request storage permissions
+- [Send data with sharesheet](user-interface/share-sender/src/main/java/com/example/platform/ui/share/sender/ShareSender.kt):
+Send texts and images with other apps using the Android Sharesheet.
```
[the remark this answers is not in the document]
Comment (yaraki) [reply samples/README.md:92]: Done

File: samples/user-interface/share-sender/README.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+- Image
+- Multiple images
+
+Android Sharesheet also provides these customization options that cover all the practical use cases
+you might need in your app.
+
+- Add or reorder targets
+- Know which app has received the data
+- Exclude target candidates
+- Add custom actions
```
[the remark this answers is not in the document]
Comment (yaraki) [reply samples/user-interface/share-sender/README.md:25]: Done. It's a good idea.

File: samples/user-interface/share-sender/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,29 @@
+
```
[the remark this answers is not in the document]
Comment (yaraki) [reply samples/user-interface/share-sender/build.gradle.kts:1]: Done

File: samples/user-interface/share-sender/src/main/java/com/example/platform/ui/share/sender/AssetFileProvider.kt [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        val segments = uri.pathSegments
+        return when (segments[0]) {
+            "image" -> MimeTypeMap.getSingleton().getMimeTypeFromExtension("jpg")
+            else -> "application/octet-stream"
+        }
+    }
+
+    override fun openAssetFile(uri: Uri, mode: String): AssetFileDescriptor? {
+        val segments = uri.pathSegments
+        println("openAssetFile: $uri")
```
[the remark this answers is not in the document]
Comment (yaraki) [reply samples/user-interface/share-sender/src/main/java/com/example/platform/ui/share/sender/AssetFileProvider.kt:49]: Done

File: samples/user-interface/share-sender/src/main/java/com/example/platform/ui/share/sender/AssetFileProvider.kt [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
+import android.content.ContentProvider
+import android.content.ContentValues
+import android.content.res.AssetFileDescriptor
+import android.database.Cursor
+import android.net.Uri
+import android.webkit.MimeTypeMap
+import androidx.core.net.toUri
+
+class AssetFileProvider : ContentProvider() {
```
[the remark this answers is not in the document]
Comment (yaraki) [reply samples/user-interface/share-sender/src/main/java/com/example/platform/ui/share/sender/AssetFileProvider.kt:27]: Done

**Pull request merged**